platform-sdk/docs/base/logging/logging.md

```diff
@@ -0,0 +1,172 @@
+
+```properties
+# Marker filter configuration
+logging.marker.CONFIG = ENABLED
+logging.marker.CRYPTO = DISABLED
+logging.marker.OTHER = DEFAULT
+```
+
+In this configuration, the `logging.marker.NAME` pattern is used, where `NAME` represents the marker's name, and the value can be set to `ENABLED`, `DISABLED`, or `DEFAULT`. For example, `logging.marker.CONFIG = ENABLED` would ensure that all log messages tagged with the `CONFIG` marker are displayed, irrespective of their log level.
+
+A typical logging configuration file, incorporating these marker filters, might look like this:
+
+```properties
+# General logging level configuration
+logging.level = INFO
+logging.level.com.swirlds.common.crypto = DEBUG
+logging.level.com.swirlds.common.crypto.Signature = WARN
+logging.level.com.hashgraph = WARN
+
+# Marker-specific logging configuration
+logging.marker.CONFIG = ENABLED
+
+# Additional handler configuration
+logging.handler.NAME.level = WARN
+```
+
+To further refine our logging system, we have incorporated the concept of handlers. These handlers allow for more granular control over logging behavior. Each handler can be distinctly named and configured using the prefix `logging.handler.NAME`, where `NAME` is a unique identifier for the handler. This structure enables us to apply handler-specific settings. For instance, `logging.handler.NAME.level` can be used to set the logging level for a specific handler. All previously discussed features, such as marker filters and log level settings, are also applicable to these handlers.
+
+A particularly useful feature of handlers is the `inheritLevels` property. This boolean property determines whether the handler should inherit the default level configurations set globally. By default, `inheritLevels` is set to `true`. However, it can be turned off to create a handler that focuses exclusively on a specific aspect of logging, such as a particular marker. For example, if you need a handler that only logs entries marked with the `CRYPTO` marker, your configuration would look like this:
+
+```properties
+# Handler specific for CRYPTO marker
+logging.handler.CRYPTO_FILE.level = OFF
+logging.handler.CRYPTO_FILE.marker.CRYPTO = ENABLED
+```
+
+In this configuration, the `CRYPTO_FILE` handler is set to ignore the global log level settings (`level = OFF`) but is specifically enabled to log messages tagged with the `CRYPTO` marker (`marker.CRYPTO = ENABLED`). This setup allows for the creation of dedicated log files or outputs for specific types of log messages, providing a focused view that can be particularly useful in complex systems or during specific types of analysis.
```

Comment:
`the creation of dedicated log files or outputs for specific types of log messages, providing a focused view`

Totally unrelated to this PR, and I'm not sure how exactly this is implemented today, but it seems that this is happening directly in the logger implementation somewhere. So I'm guessing that the code calling the logger might be penalized in terms of the performance to support all these features. It would be nice if these additional mechanisms (unconditional logging based on markers, redirecting some logs to additional destinations, etc.) would be happening in parallel (like on a different thread) or even completely offline to avoid performance degradation associated with logging which might be pretty high because in the end of the day the logging involves I/O.

If that already is the case, then it would be nice to mention this in the document to underscore that all these features don't incur any performance penalties.